editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5996,7 +5996,16 @@ void CanvasItemEditorViewport::_create_preview(const Vector<String> &files) cons
 			} else {
 				if (scene.is_valid()) {
 					Node *instance = scene->instance();
+#ifndef _MSC_VER
+#warning TODO find a better method for pausing preview
+#endif
 					if (instance) {
+						for (int j = 0; j < instance->get_child_count(); j++) {
+							Node *curr_child = instance->get_child(j);
+							if (curr_child->get_class() == "Timer") {
```

Comment:
The class checks in source are usually done like this:
```
Timer *timer = Object::cast_to<Timer>(instance->get_child(j));
if (timer)
```
This also allows you to call `stop()` directly.

Reply:
I'll keep it in mind, thank you!
What I don't currently like with the way I handled this is that it works only for timers that are immediate children of the scene's root node. IMO, a better way would be to set the scene's pause state (by using something like `instance->get_tree()->set_pause(true)`) but as I stated on the issue Godot crashes when calling this, and I need to find out why.

Reply:
You could also recursively stop the Timers, but just sounds like overkill. IMO in most cases stopping immediate Timers should be enough, the issue is not common anyways.